### PR TITLE
Several cleanups and factorization in scheme declarations

### PR DESCRIPTION
**Pull request by ppedrot**
Judging that #9838 is stalling, I am proposing a much smaller set of tweaks to the scheme declaration mechanism. It's mostly factorization and small invariants that make the code slightly more amenable to future major cleanups.

### CODE REVIEW

File: tactics/ind_tables.ml [hunk 1]
```diff
@@ -158,3 +158,5 @@ let define_mutual_scheme kind mode names mind =
 let check_scheme kind ind =
   try let _ = find_scheme_on_env_too kind ind in true
   with Not_found -> false
+
+let lookup_scheme = DeclareScheme.lookup_scheme
```
Comment (ejgallego) [inline tactics/ind_tables.ml]: why this extra fn?

File: tactics/ind_tables.ml [hunk 2]
```diff
@@ -107,7 +103,7 @@ let define_individual_scheme_base kind suff f mode idopt (mind,i as ind) =
     | Some id -> id
     | None -> add_suffix mib.mind_packets.(i).mind_typename suff in
   let role = Evd.Schema (ind, kind) in
-  let const, neff = define mode role id c (Declareops.inductive_is_polymorphic mib) ctx in
+  let const, neff = define (mode == InternalTacticRequest) role id c (Declareops.inductive_is_polymorphic mib) ctx in
```
Comment (ejgallego) [inline tactics/ind_tables.ml]: why the use of `==` ? you could also do `let internal = .... in ` as to have the code a bit more clearer.

File: tactics/ind_tables.ml (hunk 1 above)
Comment (ppedrot) [reply tactics/ind_tables.ml]: The `kind` type is hidden by the API.

File: tactics/ind_tables.ml (hunk 2 above)
Comment (ppedrot) [reply tactics/ind_tables.ml]: OK.

**Review by ejgallego — APPROVED** [review]
Cleanups look mostly good; there are still a quite a few oddities in the way thing are handled, for example declaring an inductive does use `declare_private_constant` and in general the path in declare seems a bit messy.

**Comment by ppedrot** [comment]
@ejgallego tweaked. Rereading the code, I think there is much room for improvement to get rid of recursive calls to scheme declaration functions. Most of them are already guarded with a clause for absence of scheme, and we could handle this at the level of the vernac command instead. Will submit another PR later on.

**Comment by ejgallego** [comment]
> Judging that #9838 is stalling,

Regarding #9838 unfortunately I think there is still quite a bit of work in the declare path before I'm happy to proceed more there.

**Review by SkySkimmer — APPROVED** [review]
looks good

**Pull request merged**